src/databricks/labs/ucx/azure/access.py

```diff
@@ -98,28 +99,33 @@ def _get_role_privilege(self, role_assignment: AzureRoleAssignment) -> Privilege
             privilege = self._levels[role_assignment.role_name]
         return privilege
 
-    def _map_storage(self, storage: StorageAccount) -> list[StoragePermissionMapping]:
+    def _map_storage(self, storage: StorageAccount) -> ValuesView[StoragePermissionMapping]:
         logger.info(f"Fetching role assignment for {storage.name}")
-        out = []
+        principal_spm_mapping: dict[str, StoragePermissionMapping] = {}
         for container in self._azurerm.containers(storage.id):
             for role_assignment in self._azurerm.role_assignments(str(container)):
+                # Skip the role assignments that already have WRITE_FILES privilege
+                spm_mapping_key = f"{container.container}_{role_assignment.principal.client_id}"
```

Comment:
This could cause a collision when two storage accounts have the containers with the same name. 

``` suggestion
                spm_mapping_key = f"{container}_{role_assignment.principal.client_id}"
```

Reply:
@JCZuurmond The _map_storage() method is called for every storage account separately and the dict principal_spm_mapping gets refreshed for every storage account. The dict will only contain container names for one storage account at a time.